src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java

```diff
@@ -188,14 +199,19 @@ private PodSpec createPodSpec(String appId, AppDeploymentRequest request, int po
 			podSpec.addNewImagePullSecret(properties.getImagePullSecret());
 		}
 
-		Container container = containerFactory.create(appId, request, port);
+		String countProperty = request.getDeploymentProperties().get(COUNT_PROPERTY_KEY);
+		int count = (countProperty != null) ? Integer.parseInt(countProperty) : 1;
 
-		// add memory and cpu resource limits
-		ResourceRequirements req = new ResourceRequirements();
-		req.setLimits(deduceResourceLimits(properties, request));
-		container.setResources(req);
+		for (int index = 0; index < count; index++) {
+			Container container = containerFactory.create(appId, request, port, index);
 
-		podSpec.addToContainers(container);
+			// add memory and cpu resource limits
+			ResourceRequirements req = new ResourceRequirements();
```

Comment:
I believe `ResourceRequirement` would be same for all the instances for the given app and hence the `requirement` can be computed outside the loop.